src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php

```diff
@@ -802,7 +802,7 @@ public function mergeConstraintsFrom(Builder $from)
         $wheres = $from->getQuery()->from !== $this->getQuery()->from
             ? $this->requalifyWhereTables(
                 $from->getQuery()->wheres,
-                $from->getQuery()->from,
+                $from->getQuery()->from instanceof Expression ? $from->getQuery()->from->getValue($from->getQuery()->grammer) : $from->getQuery()->from,
```

Comment:
The `getValue()` method must be called on the grammar with the value to get. The correct approach would be:

```php
$from->getQuery()->grammer->getValue($from->getQuery()->from)
```

Reply:
Oh! Thank you. I had referred to other locations in Builder which ran against Expressions. Also realized this actually had a typo and should be `grammar` and not `grammer`!